### PR TITLE
2428 Enhance docs for pure PyTorch loss, optimizer, etc

### LINKED ISSUE #2428: Enhance technical highlights for the pure PyTorch components
**Is your feature request related to a problem? Please describe.**
Got the feedback from Kristopher Kersten:
can any pytorch optimizer be used with MONAI? or does it have to be implemented in the monai.optimizers class?
working with Rhino to convert one of their existing models that uses torch.optim's SGD, and they're not exactly sure how to translate to the Novograd optimizer that's in monai.optimizer, hoping they might just be able to use the torch sgd.
Is this documented somewhere in the MONAI docs?  It wasn't obvious to Rhino, and now that I'm digging in a bit trying to help them it also seems not that obvious.  Or maybe this is just assumed since it's all based on Pytorch.

I will enhance our highlights doc to make it clear.

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #2428 .

### Description
This PR enhanced the docs to make it clear that MONAI loss, optimizer just follow the regular pure PyTorch pattern and MONAI workflows work with pure PyTorch components.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Review by wyli — APPROVED** [review]
thanks
